scripts/2.TCGA-MLexample_Pathway.py

```diff
@@ -0,0 +1,535 @@
+from sklearn.linear_model import SGDClassifier
+from sklearn.cross_validation import train_test_split
+from sklearn.metrics import roc_auc_score, roc_curve
+from sklearn.pipeline import make_pipeline
+from sklearn.preprocessing import StandardScaler
+from sklearn.feature_selection import SelectKBest
+from statsmodels.robust.scale import mad
+
+
+# In[2]:
+
+from neo4j.v1 import GraphDatabase
+
+
+# In[3]:
+
+get_ipython().magic('matplotlib inline')
+plt.style.use('seaborn-notebook')
+
+
+# ## Specify model configuration
+
+# In[4]:
+
+REACTOMEPATHWAY = 'PC7_7459'  # "Signaling by Hippo" in Reactome
+GOPATHWAY_A = 'GO:0035329'    # "hippo signaling" in the Gene Ontology
+GOPATHWAY_B = 'GO:0035330'    # "regulation of hippo signaling" in the Gene Ontology
+
+
+# In[5]:
+
+def pathway_query(path_id, node_type='BiologicalProcess', particip='GpBP'):
+    query = '''
+    MATCH (node:{})-[:PARTICIPATES_{}]-(gene)
+    WHERE node.identifier = "{}"
+    RETURN
+      gene.identifier as entrez_gene_id,
+      gene.name as gene_symbol
+    ORDER BY gene_symbol
+    '''.format(node_type, particip, path_id)
+    return query
+
+
+# In[6]:
+
+query_r = pathway_query(REACTOMEPATHWAY, node_type='Pathway', particip='GpPW')
+query_go_a = pathway_query(GOPATHWAY_A)
+query_go_b = pathway_query(GOPATHWAY_B)
+queries = [query_r, query_go_a, query_go_b]
+
+
+# In[7]:
+
+driver = GraphDatabase.driver("bolt://neo4j.het.io")
+full_results_df = pd.DataFrame()
+with driver.session() as session:
+    for query in queries:
+        result = session.run(query)
+        result_df = pd.DataFrame((x.values() for x in result), columns=result.keys())
+        full_results_df = full_results_df.append(result_df, ignore_index=True)
+classifier_genes_df = full_results_df.drop_duplicates().sort_values('gene_symbol').reset_index(drop=True)
+classifier_genes_df['entrez_gene_id'] = classifier_genes_df['entrez_gene_id'].astype('str')
+
+
+# In[8]:
+
+# Here are the genes that participate in the Hippo signaling pathway
+classifier_genes_df
+
+
+# In[9]:
+
+# Parameter Sweep for Hyperparameters
+n_feature_kept = 8000
+param_fixed = {
+    'loss': 'log',
+    'penalty': 'elasticnet',
+}
+param_grid = {
+    'alpha': [10 ** x for x in range(-6, 1)],
+    'l1_ratio': [0, 0.05, 0.1, 0.2, 0.5, 0.8, 0.9, 0.95, 1],
+}
+
+
+# *Here is some [documentation](http://scikit-learn.org/stable/modules/generated/sklearn.linear_model.SGDClassifier.html) regarding the classifier and hyperparameters*
+
+# ## Load Data
+
+# In[10]:
+
+if not os.path.exists('data'):
+    os.makedirs('data')
+
+
+# In[11]:
+
+url_to_path = {
+    # X matrix
+    'https://ndownloader.figshare.com/files/5864859':
+        os.path.join('data', 'expression-matrix.tsv.bz2'),
+    # Y Matrix
+    'https://ndownloader.figshare.com/files/5864859':
+        os.path.join('data', 'mutation-matrix.tsv.bz2'),
+    # Clinical matrix
+    'https://ndownloader.figshare.com/files/5864859':
+        os.path.join('data', 'samples.tsv')
+}
+
```

Comment:
I think there is a copy/paste error with the URLs here. I tried running this, and it only downloaded a single file. It seems to be because the URLs (the keys of this dictionary) do not change (there is only one), and so only one key gets entered in the final dictionary that gets processed.